client/landing/stepper/declarative-flow/internals/components/step-route/test/index.tsx

```diff
@@ -0,0 +1,193 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { waitFor } from '@testing-library/react';
+import { addQueryArgs } from '@wordpress/url';
+import React, { FC, Suspense } from 'react';
+import { MemoryRouter } from 'react-router';
+import recordStepStart from 'calypso/landing/stepper/declarative-flow/internals/analytics/record-step-start';
+import {
+	Flow,
+	StepperStep,
+	StepProps,
+} from 'calypso/landing/stepper/declarative-flow/internals/types';
+import { useIntent } from 'calypso/landing/stepper/hooks/use-intent';
+import { useSelectedDesign } from 'calypso/landing/stepper/hooks/use-selected-design';
+import { recordPageView } from 'calypso/lib/analytics/page-view';
+import {
+	getSignupCompleteFlowNameAndClear,
+	getSignupCompleteStepNameAndClear,
+} from 'calypso/signup/storageUtils';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { renderWithProvider } from 'calypso/test-helpers/testing-library';
+import StepRoute from '../';
+import { NavigationControls } from '../../../types';
+
+jest.mock( 'calypso/signup/storageUtils' );
+jest.mock( 'calypso/state/current-user/selectors' );
+jest.mock( 'calypso/landing/stepper/declarative-flow/internals/analytics/record-step-start' );
+jest.mock( 'calypso/landing/stepper/hooks/use-intent' );
+jest.mock( 'calypso/landing/stepper/hooks/use-selected-design' );
+jest.mock( 'calypso/lib/analytics/page-view' );
+
+const RegularStep: StepperStep = {
+	slug: 'some-step-slug',
+	asyncComponent: jest.fn(), // For the purpose of this test, we don't need to render the using lazy strategy
+};
+
+const RequiresLoginStep: StepperStep = {
+	...RegularStep,
+	requiresLoggedInUser: true,
+};
+
+const FakeStep: FC< StepProps > = () => {
+	return <div>Step Content</div>;
+};
+
+const FakeFlow: Flow = {
+	name: 'some-flow',
+	useSteps: () => [ RegularStep ],
+	useStepNavigation: () => ( {
+		submit: () => {},
+	} ),
+
+	isSignupFlow: false,
+};
```

Comment:
Minor note: can we use either `ALL_CAPS` names or a lower-case first letter for the non-component constants? At a glance, it's hard to determine that `RegularStep`, `RequiresLoginStep`, and `FakeFlow` are constants, but `FakeStep` is a component.